shell/platform/common/accessibility_bridge_unittests.cc

```diff
@@ -66,7 +66,74 @@ TEST(AccessibilityBridgeTest, basicTest) {
   EXPECT_EQ(child2_node->GetName(), "child 2");
 }
 
-TEST(AccessibilityBridgeTest, canFireChildrenChangedCorrectly) {
+// Flutter used to assume that the accessibility root had ID 0.
+// In a multi-view world, each view has its own accessibility root
+// with a globally unique node ID.
+TEST(AccessibilityBridgeTest, AccessibilityRootId) {
+  std::shared_ptr<TestAccessibilityBridge> bridge =
+      std::make_shared<TestAccessibilityBridge>();
+
+  std::vector<int32_t> children{456, 789};
+  FlutterSemanticsNode root = CreateSemanticsNode(123, "root", &children);
+  FlutterSemanticsNode child1 = CreateSemanticsNode(456, "child 1");
+  FlutterSemanticsNode child2 = CreateSemanticsNode(789, "child 2");
+
+  bridge->AddFlutterSemanticsNodeUpdate(&root);
+  bridge->AddFlutterSemanticsNodeUpdate(&child1);
+  bridge->AddFlutterSemanticsNodeUpdate(&child2);
+  bridge->CommitUpdates();
+
+  auto root_node = bridge->GetFlutterPlatformNodeDelegateFromID(123).lock();
+  auto child1_node = bridge->GetFlutterPlatformNodeDelegateFromID(456).lock();
+  auto child2_node = bridge->GetFlutterPlatformNodeDelegateFromID(789).lock();
+  auto fake_delegate = bridge->GetFlutterPlatformNodeDelegateFromID(0).lock();
+
+  EXPECT_EQ(root_node->GetChildCount(), 2);
+  EXPECT_EQ(root_node->GetData().child_ids[0], 456);
+  EXPECT_EQ(root_node->GetData().child_ids[1], 789);
+  EXPECT_EQ(root_node->GetName(), "root");
+
+  EXPECT_EQ(child1_node->GetChildCount(), 0);
+  EXPECT_EQ(child1_node->GetName(), "child 1");
+
+  EXPECT_EQ(child2_node->GetChildCount(), 0);
+  EXPECT_EQ(child2_node->GetName(), "child 2");
+
+  ASSERT_FALSE(fake_delegate);
+}
+
+// Semantic nodes can be added out of tree order.
+TEST(AccessibilityBridgeTest, AddOrder) {
+  std::shared_ptr<TestAccessibilityBridge> bridge =
+      std::make_shared<TestAccessibilityBridge>();
+
+  std::vector<int32_t> children{456, 789};
+  FlutterSemanticsNode root = CreateSemanticsNode(123, "root", &children);
+  FlutterSemanticsNode child1 = CreateSemanticsNode(456, "child 1");
+  FlutterSemanticsNode child2 = CreateSemanticsNode(789, "child 2");
+
+  bridge->AddFlutterSemanticsNodeUpdate(&child2);
+  bridge->AddFlutterSemanticsNodeUpdate(&root);
+  bridge->AddFlutterSemanticsNodeUpdate(&child1);
+  bridge->CommitUpdates();
```

Comment:
`CommitUpdates` must send updates to the `AXTree` in tree order, and, it must set the `AXTree`'s root. It sorts updates by creating sub-lists, each of which are in tree order. This test results in two sub-lists:

1. child 2
2. root, child 1